build.sbt

```diff
@@ -89,7 +89,8 @@ val plugin = project.settings(
   Compile / unmanagedSourceDirectories ++= {
     val extraDirectoriesWithPredicates = Map[String, String => Boolean](
       ("scala-3.0.x", (_.startsWith("3.0"))),
-      ("scala-3.1.0+", (v => v.startsWith("3") && !v.startsWith("3.0")))
```

Comment:
my motivation was that it's impossible for both to be true :P

Reply:
As far as I can see this would've matched on any 3.x version where x isn't 0